### DISCUSSION TITLE
All content has 'page' kind?

### DISCUSSION THREAD

**Discussion by metbril** [jpanther/congo#375] [Q&A] [answered]
Hi, as per default Hugo specs, I was expecting that the page kind `{{ .Page.Kind }}` for different types of content, would also differ.

However, I discovered that any file renders to a kind of 'page'.

I have these files in my content folder:

```
content/
  about.md
  contact.md
  posts/
    hello-again/
      index.md
    hello-world/
      index.md
```
 
When I use the custom footer to show the `{{ .Page.Kind }}` parameter, both `//localhost:1313/about/` and `//localhost:1313/posts/hello-world/` show 'page'.

**layouts/partials/extend-footer.html:**
```
Kind: {{ .Page.Kind }}
```

As per the Hugo docs, I was expecting the about page to show 'page' and hello world to show 'posts' (or 'post').

I would be using this, for (not) displaying the date and reading time for 'page' content:

**content/_index.md:**:
```
---
cascade:
  - _target:
      kind: page
    showDate: false
    showReadingTime: false
---
```

I have also tried

- .Page.Type
- .Page.Layout
- .Page.Section

which makes no difference.

Any help is much appreciated.

**Comment by jpanther** [comment, marked as answer]
> As per the Hugo docs, I was expecting the about page to show 'page' and hello world to show 'posts' (or 'post').

Where in the Hugo docs are you reading this? The `.Kind` variable can only be 5 possible values and all pages, regardless of their section, will be classified as `page` (see the [Hugo docs](https://gohugo.io/templates/section-templates/#page-kinds)). The `.Section` variable will return the base section of the content, which is what you seem to be looking for.

Why don't you just set your default site settings to hide the date and reading time, and then cascade within the posts directory to show these items without needing to target a specific kind?

**Comment by metbril** [comment]
> Where in the Hugo docs are you reading this? The .Kind variable can only be 5 possible values and all pages, regardless of their section, will be classified as page (see the [Hugo docs](https://gohugo.io/templates/section-templates/#page-kinds)). The .Section variable will return the base section of the content, which is what you seem to be looking for.

You are right. I should be section, I Guess. But that also does not show "posts" either. I'll also consider your suggestion.

To test this, I was debugging using the custom footer. And that showed the same value for all content types: pages and posts. And the Section, Type and Layout parameters were alle 'page' or empty.

I now know why: it's rendered as a cached partial, so only generated once at deploy.

```
{{ partialCached "extend-footer.html" . }}
```

You can check this in the deploy of my branch:

Contact page: https://hide-date--robertvanbregt-nl.netlify.app/contact/
Post: https://hide-date--robertvanbregt-nl.netlify.app/en/2022/02/12/owning-my-reading-list/

**Reply by jpanther** [reply]
> I now know why: it's rendered as a cached partial, so only generated once at deploy.

Perhaps this should be changed so that it is no longer cached, which would allow for more customisations here.

**Reply by metbril** [reply]
I'll leave this up to you. It would be nice, but is not required. For further testing I have used the comments template.

**Comment by metbril** [comment]
I want a lot of meta data shown on all content types but static pages. For now I have settled with this:

1. Set the general config for [article] in `params.yaml` for posts (most are `true`)
2. Cascade these settings in the front matter of the root `_index.md` page to disable for `type: page`

**content/_index.md**
```markdown
---
# Congo theme
cascade:
  - _target:
      type: "page"
    showDate: false
    showReadingTime: false
---
```

**Reply by metbril** [reply]
That did not work in the end. I 

1. set the global setting to show everything
2. moved all static pages to a folder `content/page`
3. added a cascade in the `_index.md` file in the page folder.

https://github.com/metbril/robertvanbregt-nl/blob/edf192c69b44619b8a4fe5255ea80c19f9231c94/config/_default/params.toml#L8

https://github.com/metbril/robertvanbregt-nl/blob/edf192c69b44619b8a4fe5255ea80c19f9231c94/content/page/_index.md
